src/libs/actions/IOU.ts

```diff
@@ -844,31 +863,29 @@ function getMoneyRequestInformation(
     };
 }
 
-/**
- * Requests money based on a distance (eg. mileage from a map)
- *
- * @param {Object} report
- * @param {Object} participant
- * @param {String} comment
- * @param {String} created
- * @param {String} [category]
- * @param {String} [tag]
- * @param {Number} amount
- * @param {String} currency
- * @param {String} merchant
- * @param {Boolean} [billable]
- * @param {Object} validWaypoints
- * @param {Object} policy - May be undefined, an empty object, or an object matching the Policy type (src/types/onyx/Policy.ts)
- * @param {Array} policyTags
- * @param {Array} policyCategories
- */
-function createDistanceRequest(report, participant, comment, created, category, tag, amount, currency, merchant, billable, validWaypoints, policy, policyTags, policyCategories) {
+/** Requests money based on a distance (eg. mileage from a map) */
+function createDistanceRequest(
+    report: OnyxTypes.Report,
+    participant: Participant,
+    comment: string,
+    created: string,
+    category: string | undefined,
+    tag: string | undefined,
+    amount: number,
+    currency: string,
+    merchant: string,
+    billable: boolean | undefined,
+    validWaypoints: WaypointCollection,
+    policy: OnyxTypes.Policy | EmptyObject | undefined,
+    policyTags: OnyxTypes.PolicyTags,
+    policyCategories: OnyxTypes.PolicyCategories,
+) {
     // If the report is an iou or expense report, we should get the linked chat report to be passed to the getMoneyRequestInformation function
     const isMoneyRequestReport = ReportUtils.isMoneyRequestReport(report);
     const currentChatReport = isMoneyRequestReport ? ReportUtils.getReport(report.chatReportID) : report;
 
-    const optimisticReceipt = {
-        source: ReceiptGeneric,
+    const optimisticReceipt: Receipt = {
+        source: ReceiptGeneric as string,
```

Comment:
It looks like the code expects it to be string, and if I console.log `optimisticReceipt` it's
`{source: '/1b96fa279dcd8a80ee9f.png', state: 'OPEN'}`

Reply:
Is it a string on all platforms?

Reply:
It also can be a number, I've updated the typing to take it into consideration.